crates/nu-parser/src/parser.rs

```diff
@@ -2678,7 +2678,7 @@ pub fn parse_unit_value<'res>(
 
     if let Some((unit, name, convert)) = unit_groups.iter().find(|x| value.ends_with(x.1)) {
         let lhs_len = value.len() - name.len();
-        let lhs = strip_underscores(value[..lhs_len].as_bytes());
+        let lhs = strip_underscores(&value.as_bytes()[..lhs_len]);
```

Comment:
Wow that lint is a nice little detail, never thought about it but makes total sense https://rust-lang.github.io/rust-clippy/master/index.html#sliced_string_as_bytes